google/ingester.go

```diff
@@ -0,0 +1,327 @@
+				// importing and using so we can filter by them
+				if region == "global" {
+					region = ing.region
+				}
+
+				// We want to only import the Prices/Products from the
+				// region we are importing
+				if region != ing.region {
+					continue
+				}
+
+				prod := &product.Product{
+					Provider: ProviderName,
+					SKU:      sku.SkuId,
+					Service:  service,
+					Family:   family,
+					Location: region,
+					Attributes: map[string]string{
+						"group": group,
+						"usage": usage,
+					},
+				}
+
+				// We need to set the machineFamily
+				if group == cpu || group == ram {
+					mf := strings.ToLower(strings.Split(sku.Description, " ")[0])
+					_, ok := machineFamilies[mf]
+					if ok {
+						prod.Attributes["machine_family"] = mf
+					}
+				}
+				if len(pi.TieredRates) == 0 {
+					continue
+				}
+				pwp := &price.WithProduct{
+					Price: price.Price{
+						Unit:     pi.UsageUnit,
+						Value:    decimal.NewFromFloat(float64(pi.TieredRates[0].UnitPrice.Units) + (float64(pi.TieredRates[0].UnitPrice.Nanos) / float64(1_000_000_000))),
+						Currency: pi.TieredRates[0].UnitPrice.CurrencyCode,
+					},
+					Product: prod,
+				}
+				if mf, ok := prod.Attributes["machine_family"]; ok {
+					machinteTypePrices[fmt.Sprintf("%s.%s", mf, group)] = pwp.Price
+				}
+				if ing.ingestionFilter(pwp) {
+					results <- pwp
+				}
+			}
+		}
+
+		for mt := range ing.fetchMachineTypes(ctx) {
+			mf := strings.ToLower(strings.Split(mt.Name, "-")[0])
+			_, ok := machineFamilies[mf]
+			if !ok {
+				continue
+			}
+
+			prod := &product.Product{
+				Provider: ProviderName,
+				SKU:      fmt.Sprintf("machinte-types-%d", mt.Id),
+				Service:  "Compute Engine",
+				Family:   "Compute",
+				Location: ing.region,
+				Attributes: map[string]string{
+					"machine_type":   mt.Name,
+					"group":          "MachineType",
+					"cpu":            strconv.Itoa(int(mt.GuestCpus)),
+					"ram":            strconv.Itoa(int(mt.MemoryMb)),
+					"kind":           mt.Kind,
+					"machine_family": mf,
+				},
+			}
+
+			rp, ok := machinteTypePrices[fmt.Sprintf("%s.%s", mf, ram)]
+			if !ok {
+				continue
+			}
+
+			cp, ok := machinteTypePrices[fmt.Sprintf("%s.%s", mf, cpu)]
+			if !ok {
+				continue
+			}
+
+			// The mt.MemboryMb is in MB and the rp.Unit is on GiBy
+			// so we have to convert it to the same unit
+			rp.Value = rp.Value.Mul(decimal.NewFromInt(mt.MemoryMb / 1_000))
+			cp.Value = cp.Value.Mul(decimal.NewFromInt(mt.GuestCpus))
+
+			// The Unit for the RAM is in GiBy.h, but for the addition
+			// we need the same unit which in this case would be h
+			if rp.Unit == "GiBy.h" {
+				rp.Unit = "h"
+			} else {
+				// If we cannot check the unit we skip it
+				continue
+			}
+			err := rp.Add(cp)
+			if err != nil {
+				ing.err = err
+				return
+			}
+
+			pwp := &price.WithProduct{
+				// r has the end result of the Sum of the 2 prices
```

Comment:
rp* ?

Reply:
Yeh I changed the name of the var and forgot this doc haha.